daffodil-propgen/src/main/resources/org/apache/daffodil/xsd/dafext.xsd

```diff
@@ -101,6 +101,17 @@
   <xs:element name="tunables">
     <xs:complexType>
       <xs:all>
+        <xs:element name="allowBigIntegerBits" type="xs:boolean" default="true" minOccurs="0">
+          <xs:annotation>
+            <xs:documentation>
+              Daffodil allows the use of lengthUnits="bits" when the type is xs:integer and
+              xs:nonNegativeInteger even though this is prohibited by the specification. When
+              this tunable is true, a deprecation warning is emitted when this condition is
+              encountered. When this tunable is false, this condition will instead result in a
+              schema definition error.
```

Comment:
We've changed the check from disallowing 2 indefinite size integer types to allowing only 9 definite size integer types which means that users will get warnings or errors for many more types (float, double, etc.).  I think it's important to reword this documentation so that users will know how much this tunable will impact their schemas.  Remember Daffodil used to let many schema writers pick one length unit (bits or bytes) and write every type's length using that unit without any schema diagnostics, so the new SDE/SDW diagnostic will be a big change in behavior for users.  Consider saying something like:

```
Previous Daffodil releases let schemas define every type's length using "bits" as the length unit
even though the specification allows bit length units only for a specific set of types' binary 
representations and does not allow bit length units for any other type's binary representation 
or any type's text representation.  When this tunable is true, a deprecation warning is issued
when bit length units are incorrectly used.  When this tunable is false, a schema definition
error will be issued instead.
```

Reply:
There is a separate JIRA ticket for making SDW warnings include instructions on how to suppress them (mention the specific symbol needed to put into the config file). That will help with this issue. Also.

This ticket or any that increases warnings users get, will need a release note. Probably generic one about how this release of Daffodil issues more warnings about non-portable aspects of DFDL usage, .... and how to suppress them.

Reply:
Speaking of which, we need a place to keep track of deprecation descriptions so that we remember to actually put them in the release notes.

One approach is if a commit causes deprecation/compatibility issues then there must be a clearly labeled section at the end of the commit message about what was deprecated with a specific keyword. For example, such a commit message might look like:

```
Make some non-backwards compatible/deprecation change

Description of what this commit does and why. This is more focused
on the actual implementation details of the change, like a normal
commit message.

Deprecation/Compatibility:

Message that can be copy/pasted into the release notes in the
deprecation/compatibility section. This is focused on what was
deprecated or causes backwards compatibility  issues and the
migration details for users to resolve those issues.

DAFFODIL-XYZ
```

This way when the release manager is generating the release notes they can just look at the git log for commits with the "Deprecation/Compatibility" keyword and copy the message without really having to think too much.

Alternatively we could add something like that to a comment in the ticket, but it I find searching with JIRA can be a little hard.

Reply:
Documentation has been updated